src/social_network_link_prediction/similarity_methods/global_similarity/shortest_path.py

```diff
@@ -0,0 +1,17 @@
+import networkx as nx
+from scipy.sparse import csr_matrix, lil_matrix
+
+def shortest_path(G: nx.Graph, cutoff: int = None) -> csr_matrix:
+    dim = len(G.nodes)
+    if cutoff is None:
+        cutoff = dim
+
+    lengths = dict(nx.all_pairs_shortest_path_length(G, cutoff))
+    
+    S = lil_matrix((dim, dim))
+    for source_node in lengths.keys():
+        for dest_node in lengths[source_node].keys():
+            S[source_node, dest_node] = -lengths[source_node][dest_node]
+
+    #andrebbero normalizzati i valori per riportarli tra 0 e 1? o lasciamo i negativi?
```

Comment:
Il paper che dice ?
Dato che non sono probabilità, penso che basti solo ordinarli e prendere i più alti per fare link prediction. I negativi sono solo nodi estremamente differenti tra di loro 🤔 

Al massimo puoi mettere un parametro `normalize = False` per effettuare questi calcoli se richiesto

Reply:
Si potrebbe essere sensato, inoltre ho dei dubbi sul nome dei nodi nel caso in cui non siano numeri che partono da 0

Reply:
Si, hai ragione.
Penso che con nodi che non partano da 0 si rompe.

Abbiamo la funzione che mappa ogni nodo con un indice che parte da 0. Dovrebbe risolvere il problema e non essere molto difficile da aggiungere 🤔